cert.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/TykTechnologies/tyk/certs"
+	"github.com/TykTechnologies/tyk/config"
+
+	"github.com/gorilla/mux"
+)
+
+type APICertificateStatusMessage struct {
+	CertID  string `json:"id"`
+	Status  string `json:"status"`
+	Message string `json:"message"`
+}
+
+type APIAllCertificates struct {
+	CertIDs []string `json:"certs"`
+}
+
+// dummyGetCertificate needed because TLSConfig require setting Certificates array or GetCertificate function from start, even if it get overriden by `getTLSConfigForClient`
+func dummyGetCertificate(*tls.ClientHelloInfo) (*tls.Certificate, error) {
+	return nil, nil
+}
+
+func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *tls.ClientHelloInfo) (*tls.Config, error) {
+
+	// Supporting legacy certificate configuration
+	serverCerts := make([]tls.Certificate, 0)
+	certNameMap := make(map[string]*tls.Certificate)
```

Comment:
Since you're using `make`, you might as well init the `cap`s to `len(config.Global.HttpServerOptions.Certificates)`

Reply:
It is possible but note that this is a static code which runs only on gateway start.

Reply:
Sure, but I was just thrown off by your explicit setting of `0` as the capacity. If you don't care about performance, you can omit it or use `[]tls.Certificate{}` and `map[string]*tls.Certificate{}`.